CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.6)
+
+if (DEFINED ENV{VCPKG_ROOT})
+  set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake")
```

Comment:
Toolchain files are evaluated very early, even before all the compiler checks. Setting this variable has no effect at all, it is supposed to be passed on the command line.

Did you mean to `include()` this file instead?

Reply:
This is not my PR, but honestly I'm still very much against adding automagic Vcpkg detection for multiple reasons:

- It cannot be turned off. If I wanted to pick up an OpenCL version other than what ships with Vcpkg it is impossible if it's hardcoded in the script.
- It is unorthodox. It goes against the biggest feature of Vcpkg, that scripts need not know of it's existence. That is the primary reason why Vcpkg is better than Conan. Scripts need to be Conan aware, while using Vcpkg is transparent to downstreams.

Users of Vcpkg already have a way of not spamming the command-line with toolchain file specification. If they use an IDE, in VS the toolchain file can be set in CMakeSettings.json ([now with GUI](https://blogs.msdn.microsoft.com/vcblog/2019/01/24/introducing-the-new-cmake-project-settings-ui/)) in a set-it-and-forget it manner once and for all. VS Code users are probably using [CMake Tools](https://marketplace.visualstudio.com/items?itemName=vector-of-bool.cmake-tools) and set the [toolchain file via Kits](https://vector-of-bool.github.io/docs/vscode-cmake-tools/kits.html#toolchain), again: set-it-and-forget-it. Even if someone is a console junkie on Linux (and not using VS Code on Linux), they probably got tired of `-D CMAKE_TOOLCHAIN_FILE=$(VCPKG_ROOT)/scripts/buildsystems/vcpkg.cmake ...` so they most likely have an alias set that adds this one variable to the invocation.

Long story short, I don't see the benefit of this, other than alienating non-Vcpkg users of ever reyling on it by giving the impression the scripts need to use this or just simply plaguing projects with it. Removing makes the script shorter, leaner and nicer in every regard.

Reply:
I don't mind either way, but you can always override the default search paths for a package by passing `-DOpenCL_ROOT=C:/myopencl`. But speaking as a Conan user, I don't like the idea of using `CMAKE_TOOLCHAIN_FILE` to inject package management like this. How does cross-compilation work with vcpkg?

> That is the primary reason why Vcpkg is better than Conan. Scripts need to be Conan aware, while using Vcpkg is transparent to downstreams.

This is OT, but Conan's [cmake_path generator](https://docs.conan.io/en/latest/integrations/cmake/cmake_paths_generator.html) integrates exactly like vcpkg.

Reply:
>How does cross-compilation work with vcpkg?

Either our toolchain can be pulled in at the bottom of your own toolchain, or you can ask us to chainload yours via `-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=xyz`.